File: src/main/java/io/github/hiiragi283/api/shape/HTShape.java
package io.github.hiiragi283.api.shape;

import java.util.Collection;
import java.util.stream.Stream;

import com.github.bsideup.jabel.Desugar;

@Desugar
public record HTShape(HTShapeKey key, Collection<HTShapeItemBuilder> itemBuilders) {

    public Stream<HTShapeItemBuilder> ItemBuilderStream() {
        return itemBuilders.stream();
    }
}

File: src/main/kotlin/io/github/hiiragi283/api/shape/HTShape.kt
package io.github.hiiragi283.api.shape

import io.github.hiiragi283.api.material.HTMaterialKey
import io.github.hiiragi283.api.property.HTPropertyHolder

fun interface HTShape {
    fun canGenerateItem(materialKey: HTMaterialKey, material: HTPropertyHolder): Boolean
}